app/helpers/application_helper/button/vm_native_viewer_console.rb

```diff
@@ -0,0 +1,16 @@
+class ApplicationHelper::Button::VmNativeViewerConsole < ApplicationHelper::Button::Basic
+  needs :@record
+
+  def visible?
+    @record.vendor == 'redhat'
+  end
+
+  def disabled?
+    begin
+      @record.validate_native_viewer_support
```

Comment:
Is this calling something from the provider's API, because then it can heavily affect the performance. cc @agrare

Reply:
Oh sorry, didn't see the wip :confused:

Reply:
@skatemsn this call is similar to the one that is doing vmware VRMC button, it's only to test if ems is not nil and state is 'on'. No call to external provider.

Reply:
please feel free to suggest modifications

Reply:
BTW Merry Christmas 'cause I realized that midnight has passed

Reply:
Okkay, I'm gonna dig into these. Happy new year!

Reply:
Many thanks and Happy new year also to you and the team